addons/binding/org.openhab.binding.tado/src/main/api/pom.xml

```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
Imho, this is a strange place for a pom.xml

@pfink @kaikreuzer Can we just place this (maybe with a specific profile) in the normal bindings pom.xml or will that give conflicts?

Reply:
I thought about that, but couldn't find a clean way of doing it. What the pom file does, is
- generate sources
- compile them
- package to a jar in lib folder

And it should not be executed automatically.

By integrating it into the binding's pom file, we would have to generate 2 jar files from the same pom file, one for the binding, the other one for the api client. And we have to make sure that they are using separate set of classes. Binding classes must not end up in the api jar and vice versa.
So I've decided to put the api in a separate pom file instead of trying to handle 2 artefacts from one pom file.

I should probably move the api folder to the binding's top level directory. Now that the code is gone, `src/main` doesn't make much sense any longer.

Reply:
Hm, I do not have any good idea either - I understand @dfrommi's arguments and as a result, we can consider this pom.xml just like some .sh script in here: A tool for developers to trigger some manual code generation. So unless @pfink has some better suggestion, I'd be ok to leave it this way.

Reply:
Why not just put this into a profile?

Reply:
@dfrommi what about pfink's suggestion to use a maven-profile for this?

Reply:
@pfink would you be able to assist @dfrommi with the challenges he mentions above:

> By integrating it into the binding's pom file, we would have to generate 2 jar files from the same pom file, one for the binding, the other one for the api client. And we have to make sure that they are using separate set of classes. Binding classes must not end up in the api jar and vice versa.

Reply:
I don't see how that would solve the problem of generating 2 artefacts in one pom file, but I have to admit that my maven knowledge is rather outdated.

Let's assume we put everything that is code-generation related in a profile, then the generation is not executed unless the profile is activated. So far so good.
But when you want to generate the code, all the defaults from maven are still active, like code being in `src/main/java` that is compiled, but it should not be added to the client-jar file. To fix it, the generation part in the profile would have to disable or reconfigure the defaults and also make sure that classes from the binding are removed from target before the client-jar is generated or use another target-folder.
I'm sure all that is possible, but is this really better and easier to understand?

We might of course try it, but I personally don't see a benefit and I would also need some help.

Reply:
Just curoius is there a reason that it should be jar, can we just generate the sources in a specific directory and not create a jar?